app/src/main/java/org/connectbot/ConsoleActivity.java

```diff
@@ -985,15 +938,16 @@ protected void onNewIntent(Intent intent) {
 					return;
 				}
 
-				requestedIndex = addNewTerminalView(requestedBridge);
+				adapter.notifyDataSetChanged();
+				requestedIndex = adapter.getCount();
 			} else {
-				final int flipIndex = getFlipIndex(requestedBridge);
+				final int flipIndex = bound.getBridges().indexOf(requestedBridge);
 				if (flipIndex > requestedIndex) {
 					requestedIndex = flipIndex;
 				}
 			}
 
-			setDisplayedTerminal(requestedIndex);
+			pager.setCurrentItem(requestedIndex);
```

Comment:
I think this is also missing an `updatePromptVisible()`, etc., as above and should be wrapped in a call to `setDisplayedTerminal(…)`.